cl/citations/match_citations.py

```diff
@@ -139,3 +146,110 @@ def match_citation(citation, citing_doc=None):
 
     # Give up.
     return []
+
+
+def get_citation_matches(opinion, citations):
+    # A list of opinions, as matched to citations
+    citation_matches = []
+    was_matched = False
+
+    for citation in citations:
+        matched_opinion = None
+
+        # If the citation is a supra citation, try to resolve it to one of
+        # the citations that has already been matched
+        if isinstance(citation, SupraCitation):
+            for cm in citation_matches:
+                # The only clue we have to help us with resolution is the guess
+                # of what the supra citation's antecedent is, so we try to
+                # match that string to one of the known case names of the
+                # already matched opinions.
+                if (
+                    strip_punct(citation.antecedent_guess)
+                    in cm.cluster.case_name_full
```

Comment:
I think I'd recommend using `base_case_name` here, which is just a helper function that gives you the `case_name`, `case_name_full`, or `case_name_short` of the item, in that order. I'd be careful though about using case_name_full, because they can be obscenely long (they're also often missing, which is what `best_case_name` helps with). I think using the shorter case name as a first pass and then falling back to the full case name is probably smart. Another thought....is it likely or possible that we'll see a lot of examples like this:

> ...v. IBM Inc., supra at 23

In which case we'd pick up on "Inc," which would be a really common term? I wonder if we need a blacklist for things like this that are common in lots of case names. "State" and "States" comes to mind too. What are the odds you have "Johnson v. State" and "Hamilton v. State" in the same case? Fairly high, I'd wager.

Reply:
1) Sure, that makes sense. I wasn't actually sure which case name field was best to use. However, I don't see `base_case_name` anywhere -- is that the name of the function?
2) Yes, excellent point. For some reason, I made the (erroneous) assumption that the `antecedent_guess` for supras would never cause a match conflict. By creating a blacklist, do you mean blacklisting possible antecedents during the parsing process? (So, if we encounter a blacklisted word -- like state or inc. -- we can keep scanning in the hopes of finding a more unique one?)

Reply:
1. Sorry, typo. Should be `best_case_name`. 

2. I was thinking of a hardcoded blacklist of words like "state" and "inc" or whatever, but I'm not at all married to it. We might have a list like this in the code base already if you poke around for something like "v.", it might turn up? Sorry, I forget. 

    I guess another approach is to try matching the antecedent as we do now and if it matches more than one prior citation, give up. Working our ways backwards past common words is probably helpful too, so maybe a combination of both approaches would be safest?

Reply:
1. Fixed in c839569
2. In the same commit, I also changed the supra matching logic to give up if an antecedent guess matches two possible candidates, per your second suggestion. I think the blacklist idea is interesting, but something to add in a future PR? (Since this fix should at least avoid any problems of false matching, even if it doesn't take full advantage of all the information available.)

Reply:
Yeah, seems OK for now. I guess we'll see when we get to real data exactly how much more refinement this needs.